src/cascadia/TerminalApp/SuggestionsControl.cpp

```diff
@@ -0,0 +1,1103 @@
+            TraceLoggingKeyword(MICROSOFT_KEYWORD_MEASURES),
+            TelemetryPrivacyDataTag(PDT_ProductAndServiceUsage));
+    }
+
+    // Method Description:
+    // - Event handler for when the text in the input box changes. In Action
+    //   Mode, we'll update the list of displayed commands, and select the first one.
+    // Arguments:
+    // - <unused>
+    // Return Value:
+    // - <none>
+    void SuggestionsControl::_filterTextChanged(const IInspectable& /*sender*/,
+                                                const Windows::UI::Xaml::RoutedEventArgs& /*args*/)
+    {
+        // We're setting _lastFilterTextWasEmpty here, because if the user tries
+        // to backspace the last character in the input, the Backspace KeyDown
+        // event will fire _before_ _filterTextChanged does. Updating the value
+        // here will ensure that we can check this case appropriately.
+        _lastFilterTextWasEmpty = _searchBox().Text().empty();
+
+        _updateFilteredActions();
+
+        // In the command line mode we want the user to explicitly select the command
+        _filteredActionsView().SelectedIndex(0);
+
+        const auto currentNeedleHasResults{ _filteredActions.Size() > 0 };
+        _noMatchesText().Visibility(currentNeedleHasResults ? Visibility::Collapsed : Visibility::Visible);
+        if (auto automationPeer{ Automation::Peers::FrameworkElementAutomationPeer::FromElement(_searchBox()) })
+        {
+            automationPeer.RaiseNotificationEvent(
+                Automation::Peers::AutomationNotificationKind::ActionCompleted,
+                Automation::Peers::AutomationNotificationProcessing::ImportantMostRecent,
+                currentNeedleHasResults ?
+                    winrt::hstring{ fmt::format(std::wstring_view{ RS_(L"CommandPalette_MatchesAvailable") }, _filteredActions.Size()) } :
+                    NoMatchesText(), // what to announce if results were found
+                L"SuggestionsControlResultAnnouncement" /* unique name for this group of notifications */);
+        }
+    }
+
+    Collections::IObservableVector<winrt::TerminalApp::FilteredCommand> SuggestionsControl::FilteredActions()
+    {
+        return _filteredActions;
+    }
+
+    void SuggestionsControl::SetActionMap(const Microsoft::Terminal::Settings::Model::IActionMapView& actionMap)
+    {
+        _actionMap = actionMap;
+    }
+
+    void SuggestionsControl::SetCommands(const Collections::IVector<Command>& actions)
+    {
+        _allCommands.Clear();
+        for (const auto& action : actions)
+        {
+            auto actionPaletteItem{ winrt::make<winrt::TerminalApp::implementation::ActionPaletteItem>(action) };
+            auto filteredCommand{ winrt::make<FilteredCommand>(actionPaletteItem) };
+            _allCommands.Append(filteredCommand);
+        }
+
+        if (Visibility() == Visibility::Visible)
+        {
+            _updateFilteredActions();
+        }
+        else // THIS BRANCH IS NEW
```

Comment:
new where exactly? this whole file is new!

Reply:
Oh, that one is new relative to the command palette's code. That comment is useless though... but why did I add that?